### PR TITLE
chore: add abi_wasm publishing

### PR DESCRIPTION
**Pull request by jonybur**
# Description

Adds a workflow to manually trigger a publish for abi_wasm

## Problem\*

abi_wasm wasn't getting published to npm.

Resolves https://github.com/noir-lang/noir/issues/2582

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: .github/workflows/release-abi_wasm.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Copy Build Output to Temporary Directory
+        run: |
+          mkdir temp_publish_dir
+          cp -r ${{ env.BUILD_OUTPUT_PATH }}/* temp_publish_dir/
+
+      - name: Publish to NPM
+        working-directory: ./temp_publish_dir
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (TomAFrench) [inline .github/workflows/release-abi_wasm.yml:37]: Is this necessary? Can we not publish directly from the build output directly?

Comment (jonybur) [reply .github/workflows/release-abi_wasm.yml:37]: I think that the directory is read-only and "publish" will try to create a .tar; that's why I'm copying it over.

**Review by kobyhallx — APPROVED** [review]
LGTM

File: .github/workflows/release-abi_wasm.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Release ABI Wasm
+
+on:
+  workflow_dispatch:
+
+jobs:
+  noirc-abi-wasm-build:
+    runs-on: ubuntu-latest
+    env:
+      CACHED_PATH: /tmp/nix-cache
```
Comment (TomAFrench) [inline .github/workflows/release-abi_wasm.yml:10]:
```suggestion
    env:
      CACHED_PATH: /tmp/nix-cache
```

This is unused

**Pull request merged**